### PR TITLE
Enforce the retention policy for AspirEDU backups.

### PR DESCRIPTION
**Pull request by tim-schilling**
This should look through our past backups and remove any that are not on a day we'd like to retain.

This is a WIP

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #
-# This file is autogenerated by pip-compile with Python 3.11
+# This file is autogenerated by pip-compile with Python 3.10
```
Comment (tim-schilling) [inline requirements.txt:2]: This needs to be switched back to 3.11 ugh

File: src/delete_backups.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    pass
+
+
+def is_saturday(value: date) -> bool:
+    """Determine if date is a Saturday"""
+    return value.weekday() == 5
+
+
+@lru_cache(maxsize=None)
+def saturdays_for_the_next_year() -> set[date]:
```
Comment (tim-schilling) [inline src/delete_backups.py:35]: This should be for the past year, not next year

File: src/delete_backups.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    """Find all Saturdays between from today to 1 year"""
+    start = date.today()
+    cut_off = start - relativedelta(years=1)
+    return set(d.date() for d in rrule(WEEKLY, byweekday=SA(1), dtstart=cut_off, until=start))
+
+
+@lru_cache(maxsize=None)
+def first_saturdays_of_year1_to_year3() -> set[date]:
```
Comment (tim-schilling) [inline src/delete_backups.py:43]: Mention that this is per month, not every saturday.

**Comment by tim-schilling** [comment]
@Nish8192 this is ready for review tomorrow

File: src/delete_backups_test.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    assert saturdays[-1] == date(2018, 12, 1)
+
+
+@time_machine.travel(datetime(2020, 1, 1, tzinfo=utc_tz))
+def test_meets_retention_policy():
+    today = date.today()
+    retain_dates = list(
+        sorted((d for d in get_test_dates() if meets_retention_policy(d)), reverse=True)
+    )
+    # Do some quick checks on teh first 64 dates since they are easy
```
Comment (Nish8192) [inline src/delete_backups_test.py:45]: nitpick: `teh`, typo

**Review by Nish8192 — APPROVED** [review]

File: src/delete_backups.py [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+    start = today - relativedelta(years=1)
+    cut_off = today - relativedelta(years=3)
+    return set(
+        d.date() for d in rrule.rrule(rrule.MONTHLY, byweekday=SA(1), dtstart=cut_off, until=start)
+    )
+
+
+def meets_retention_policy(value: date):
+    today = date.today()
+    if (today - timedelta(weeks=2)) < value <= today:
```
Comment (Nish8192) [inline src/delete_backups.py:62]: A comment here to point out that the retention policy of retaining all back ups for 2 weeks is being enforced here should make it easier for someone in the future to adjust as necessary.

**Pull request merged**